pkg/controllers/subnet/subnet_controller.go

```diff
@@ -64,23 +64,20 @@ func (r *SubnetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		if !controllerutil.ContainsFinalizer(obj, servicecommon.SubnetFinalizerName) {
 			controllerutil.AddFinalizer(obj, servicecommon.SubnetFinalizerName)
 
-			if obj.Spec.AccessMode == "" || obj.Spec.IPv4SubnetSize == 0 {
-				vpcNetworkConfig := r.VPCService.GetVPCNetworkConfigByNamespace(obj.Namespace)
-				if vpcNetworkConfig == nil {
-					err := fmt.Errorf("operate failed: cannot get configuration for Subnet CR")
-					log.Error(nil, "failed to find VPCNetworkConfig for Subnet CR", "subnet", req.NamespacedName, "namespace %s", obj.Namespace)
-					updateFail(r, ctx, obj, "")
-					return ResultRequeue, err
-				}
-
-				if obj.Spec.AccessMode == "" {
-					log.Info("aobj.Spec.AccessMode set", "subnet", req.NamespacedName)
-					obj.Spec.AccessMode = v1alpha1.AccessMode(v1alpha1.AccessModePrivate)
-				}
-				if obj.Spec.IPv4SubnetSize == 0 {
-					obj.Spec.IPv4SubnetSize = vpcNetworkConfig.DefaultSubnetSize
-				}
+			if obj.Spec.AccessMode == "" {
+				log.Info("obj.Spec.AccessMode set", "subnet", req.NamespacedName)
+				obj.Spec.AccessMode = v1alpha1.AccessMode(v1alpha1.AccessModePrivate)
 			}
+
+			vpcNetworkConfig := r.VPCService.GetVPCNetworkConfigByNamespace(obj.Namespace)
+			if vpcNetworkConfig == nil {
+				err := fmt.Errorf("operate failed: cannot get configuration for Subnet CR")
+				log.Error(nil, "failed to find VPCNetworkConfig for Subnet CR", "subnet", req.NamespacedName, "namespace %s", obj.Namespace)
+				updateFail(r, ctx, obj, "")
+				return ResultRequeue, err
+			}
+
+			obj.Spec.IPv4SubnetSize = vpcNetworkConfig.DefaultSubnetSize
```

Comment:
I think you also need to check `obj.Spec.IPv4SubnetSize == 0` as user can manually create a Subnet/SubnetSet CR with size, right?

Reply:
Updated.